File: frontend/user/lib/domain/entity/product/product_detail.dart
class ProductDetail {
  final int id;
  final int colorId;
  final String colorName;
  final String colorHex;
  final int sizeId;
  final String size;
  final double price;
  final bool isActive;

  ProductDetail({
    required this.id,
    required this.colorId,
    required this.colorName,
    required this.colorHex,
    required this.sizeId,
    required this.size,
    required this.price,
    required this.isActive,
  });
}

File: backend/app/infrastructure/database/models/product_detail.py
from sqlalchemy import Boolean, Column, String, Integer, ForeignKey
from sqlalchemy.orm import relationship
from app.infrastructure.database.base import Base

class ProductDetail(Base):
    __tablename__ = "product_details"

    id = Column(Integer, primary_key=True, index=True)
    product_id = Column(Integer, ForeignKey("products.id", ondelete="CASCADE"))
    color_id = Column(Integer, ForeignKey("colors.id"))
    size_id = Column(Integer, ForeignKey("sizes.id"))
    price = Column(Integer, nullable=True)
    is_active = Column(Boolean, nullable=False, default=True)

    
    product = relationship("Product", back_populates="product_details")
    color = relationship("Color", back_populates="product_details")
    size = relationship("Size", back_populates="product_details")

    order_details = relationship("OrderDetail", back_populates="product_detail")
    cart_details = relationship("CartDetail", back_populates="product_detail")